src/gluonts/model/naive_2/_predictor.py

```diff
@@ -0,0 +1,166 @@
+    """
+    assert freq is not None or season_length is not None, (
+        "Either the frequency or season length of the time series "
+        "has to be specified. "
+    )
+    season_length = (
+        season_length if season_length is not None else get_seasonality(freq)
+    )
+    has_seasonality = False
+
+    if season_length > 1:
+        has_seasonality = seasonality_test(past_ts_data, season_length)
+
+    # it has seasonality, then calculate the multiplicative seasonal component
+    if has_seasonality:
+        # TODO: think about maybe only using past_ts_data[- max(5*season_length, 2*forecast_length):] for speedup
+        seasonal_decomposition = sm.tsa.seasonal_decompose(
+            x=past_ts_data, period=season_length, model="multiplicative"
+        ).seasonal
+        seasonality_normed_context = past_ts_data / seasonal_decomposition
+
+        last_period = seasonal_decomposition[-season_length:]
+        num_required_periods = (forecast_length // len(last_period)) + 1
+
+        multiplicative_seasonal_component = np.tile(
+            last_period, num_required_periods
+        )[:forecast_length]
+    else:
+        seasonality_normed_context = past_ts_data
+        multiplicative_seasonal_component = np.ones(
+            forecast_length
+        )  # i.e. no seasonality component
+
+    # calculate naive 1 forecast: (last value forecast_length times)
+    naive_1_forecast = (
+        np.ones(forecast_length) * seasonality_normed_context[-1]
+    )
+
+    forecast = np.mean(naive_1_forecast) * multiplicative_seasonal_component
+
+    return forecast
+
+
+class Naive2Predictor(RepresentablePredictor):
+    """
+    Naïve 2 forecaster as described in the M4 Competition Guide:
+    https://www.m4.unic.ac.cy/wp-content/uploads/2018/03/M4-Competitors-Guide.pdf
+
+    The python analogue implementation to:
+    https://github.com/Mcompetitions/M4-methods/blob/master/Benchmarks%20and%20Evaluation.R#L118
+
+    Parameters
+    ----------
+    freq
+        Frequency of the input data
+    prediction_length
+        Number of time points to predict
+    season_length
+        Length of the seasonality pattern of the input data
+    """
+
+    @validated()
+    def __init__(
+        self,
+        freq: str,
+        prediction_length: int,
+        season_length: Optional[int] = None,
+    ) -> None:
+        super().__init__(prediction_length, freq)
+
+        assert (
+            season_length is None or season_length > 0
+        ), "The value of `season_length` should be > 0"
+
+        self.freq = freq
+        self.prediction_length = prediction_length
+        self.season_length = (
+            season_length
+            if season_length is not None
+            else get_seasonality(freq)
+        )
+
+    def predict_item(self, item: DataEntry) -> Forecast:
+        start_time = pd.Timestamp(item["start"], freq=self.freq)
```

Comment:
I think `item["start"]` already is of `pd.Timestamp`.

Reply:
yeah you're right